services/accounts/settings.go

```diff
@@ -7,16 +7,19 @@ import (
 	"github.com/status-im/status-go/multiaccounts/settings"
 	"github.com/status-im/status-go/nodecfg"
 	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/protocol"
+	"github.com/status-im/status-go/protocol/identity"
 )
 
-func NewSettingsAPI(db *accounts.Database, config *params.NodeConfig) *SettingsAPI {
-	return &SettingsAPI{db, config}
+func NewSettingsAPI(messenger **protocol.Messenger, db *accounts.Database, config *params.NodeConfig) *SettingsAPI {
```

Comment:
`messenger` is already a pointer. Why is a pointer to a pointer necessary?

Reply:
Because apparently messenger points to nil just after service creation. If it was not pointer to pointer, later init would be not reflected in settings API, resulting in a null pointer dereference and crash (I was there).

The same convention is used there: https://github.com/status-im/status-go/blob/d0e0deac952aaf10d2deafc406fe62e8b552eb2e/services/accounts/accounts.go#L24

Reply:
Ah! i have run into that issue too. If possible please add a comment on why is the pointer pointer necessary.
Another alternative to the pointer pointer could be to call a `Init` method in here https://github.com/status-im/status-go/blob/develop/api/geth_backend.go#L1258-L1265 .